### PR TITLE
Document HTTP Status Code per HTTP endpoint.

### PR DESCRIPTION
**Pull request by Natim**
r? @leplatrem

### CODE REVIEW

File: docs/api/batch.rst [hunk 1]
```diff
@@ -99,6 +99,13 @@ The response body is a list of all responses:
       ]
     ]
 
+Error codes
+:::::::::::
+
+* 200 OK: The request have been processed
```
Comment (ametaireau) [inline docs/api/batch.rst]: nit: we should add ";" at the end of each option.

Comment (ametaireau) [inline docs/api/batch.rst]: nit: we should probably use ``` `200 OK` ``` so that it's formatted differently.

Comment (ametaireau) [inline docs/api/batch.rst]: I don't get why 200 ok is under "error codes". Should we rename the section to "HTTP status codes" ?

File: docs/api/resource.rst [hunk 2]
```diff
@@ -140,6 +140,16 @@ Filtering, sorting and paginating can all be combined together.
...
 
 
+HTTP Status Codes
+-----------------
+
+* ``200 OK``: The request have been processed
+* ``304 Not Modified``: Collection items did not change since ``If-Unmodified-Since`` header value
+* ``400 Bad Request``: The request body is invalid
+* ``412 Precondition Failed``: Collection items changed since provided ``If-Unmodified-Since`` header value
+* ``503 Service Unavailable``: A backend is unavailable
```
Comment (ametaireau) [inline docs/api/resource.rst]: This could be something else: http://pretty-rfc.herokuapp.com/RFC2616#status.503

**Comment by ametaireau** [comment]
I find it a little weird to repeat these over and over again (yeah, we need better documentation management).

File: docs/api/resource.rst [hunk 3]
```diff
@@ -180,6 +190,15 @@ If a conflict occurs, an error response is returned with status ``409``.
 A ``existing`` attribute in the response gives the offending record.
 
 
+HTTP Status Codes
+-----------------
+
+* ``200 OK``: The request have been processed
```
Comment (leplatrem) [inline docs/api/resource.rst]: 201 Created

File: docs/api/resource.rst [hunk 4]
```diff
@@ -196,6 +215,15 @@ If the request header ``If-Unmodified-Since`` is provided, and if the collection
 has changed meanwhile, a ``412 Precondition failed`` error is returned.
 
 
+HTTP Status Codes
+-----------------
+
+* ``200 OK``: The request have been processed
+* ``405 Method Not Allowed``: This endpoint is not available
```
Comment (leplatrem) [inline docs/api/resource.rst]: Memo: #46

File: docs/api/batch.rst [hunk 5]
```diff
@@ -99,6 +99,13 @@ The response body is a list of all responses:
       ]
     ]
 
+HTTP Status Codes
+-----------------
+
+* ``200 OK``: The request have been processed
+* ``400 Bad Request``: The request body is invalid
+* ``503 Service Unavailable``: The service is currently Unavailable.
```
Comment (leplatrem) [inline docs/api/batch.rst]: Since you mention 503, should we mention 401 and 403 ?

File: docs/api/resource.rst [hunk 6]
```diff
@@ -243,6 +287,14 @@ Validation and conflicts behaviour is similar to creating records (``POST``).
 If the request header ``If-Unmodified-Since`` is provided, and if the record has
 changed meanwhile, a ``412 Precondition failed`` error is returned.
 
+HTTP Status Code
+----------------
+
+* ``200 OK``: The request have been processed
+* ``400 Bad Request``: If the record id does not match an existing record
+* ``412 Precondition Failed``: Collection items changed since provided ``If-Unmodified-Since`` header value
+* ``503 Service Unavailable``: The service is currently Unavailable
```
Comment (leplatrem) [inline docs/api/resource.rst]: 409 Conflict ?

File: docs/api/resource.rst [hunk 7]
```diff
@@ -274,3 +326,12 @@ Conflicts
...
 ``409 Conflict`` error response is returned (see :ref:`error channel <error-responses>`).
+
+
+HTTP Status Code
+----------------
+
+* ``200 OK``: The request have been processed
+* ``400 Bad Request``: The request body is invalid
+* ``412 Precondition Failed``: Collection items changed since provided ``If-Unmodified-Since`` header value
+* ``503 Service Unavailable``: The service is currently Unavailable
```
Comment (leplatrem) [inline docs/api/resource.rst]: 409 Conflict ?

**Comment by Natim** [comment]
@leplatrem Updated

File: docs/api/resource.rst [hunk 8]
```diff
@@ -180,6 +189,15 @@ If a conflict occurs, an error response is returned with status ``409``.
 A ``existing`` attribute in the response gives the offending record.
 
 
+HTTP Status Codes
+-----------------
+
+* ``200 OK``: This record already exists, here is the one stored on the database;
```
Comment (leplatrem) [inline docs/api/resource.rst]: This behaviour is only implemented in readinglist actually :( It raises a 409 in cliquet

Comment (Natim) [inline docs/api/resource.rst]: We should be consistent on this with cliquet since as we discussed it, we are following the HTTP spec on this.

Comment (leplatrem) [inline docs/api/resource.rst]: @Natim, then we should document the way it is currently implemented, and open an issue to adress this.

Comment (Natim) [inline docs/api/resource.rst]:
```
    @resource.view(permission='readwrite')
    def collection_post(self, *args, **kwargs):
        try:
            return super(Article, self).collection_post(*args, **kwargs)
        except httpexceptions.HTTPConflict as e:
            self.request.response.status_code = 200
            return e.existing
```

Comment (Natim) [inline docs/api/resource.rst]: I have created #76 to handle that.

**Pull request merged**